File: backend/app/models/driver.py
import enum
from datetime import date, datetime
from typing import Optional

from sqlalchemy import Date, DateTime, Enum, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


class DriverStatus(str, enum.Enum):
    active = "active"
    inactive = "inactive"


class Driver(Base):
    __tablename__ = "drivers"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    full_name: Mapped[str] = mapped_column(String(255))
    identification_number: Mapped[Optional[str]] = mapped_column(String(50), nullable=True)
    phone: Mapped[Optional[str]] = mapped_column(String(50), nullable=True)
    whatsapp: Mapped[Optional[str]] = mapped_column(String(50), nullable=True)
    email: Mapped[Optional[str]] = mapped_column(String(255), nullable=True)
    driver_license_number: Mapped[Optional[str]] = mapped_column(String(50), nullable=True)
    license_expiration_date: Mapped[Optional[date]] = mapped_column(Date, nullable=True)
    authorized_vehicles: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    notes: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    status: Mapped[DriverStatus] = mapped_column(Enum(DriverStatus), default=DriverStatus.active)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())
    updated_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())

File: frontend/src/types/driver.ts
export type DriverStatus = "active" | "inactive";

export interface Driver {
  id: number;
  full_name: string;
  identification_number: string | null;
  phone: string | null;
  whatsapp: string | null;
  email: string | null;
  driver_license_number: string | null;
  license_expiration_date: string | null;
  authorized_vehicles: string | null;
  notes: string | null;
  status: DriverStatus;
  created_at: string;
  updated_at: string;
}

export interface DriverFormData {
  full_name: string;
  identification_number: string;
  phone: string;
  whatsapp: string;
  email: string;
  driver_license_number: string;
  license_expiration_date: string;
  authorized_vehicles: string;
  notes: string;
  status: DriverStatus;
}
